### ISSUE TITLE
.md files in /content not shown on the website

### ISSUE THREAD

**Issue by gelavat** [picocms/Pico#592]
Hello.

If I copy the "pico\vendor\picocms\pico\content-sample\index.md" to the "pico\content\index.md" file, and edit it, the modification is not shown.
All .md files that I put in the contents directory are not shown on the website.
also if I install a new theme, it doesn't show the index.md file at all, only the theme.
What can be the problem?

The DOM and MDSTRING php modules are enabled on the web server.

**Comment by PhrozenByte** [comment]
Pico tries to guess the content file by checking whether the file `content/index.md` exists. I assume wrong file permissions, so that your webserver / PHP process can't access the file. Check the file permissions. You can overrule Pico's autodiscorvery by setting the [`content_dir` option in your `config/config.yml`](https://github.com/picocms/Pico/blob/master/config/config.yml.template#L36) (create a new file, don't just edit the `config.yml.template`), even though I assume this won't work either due to invalid file permissions. However, you'll force Pico to use that directory and should receive an PHP error instead (usually a white/blank page, check your webserver's `error.log` resp. PHP's error log then)

**Comment by gelavat** [comment]
Thanks for your support. It was not file permissions related.
I digged into that and in the end I figured out that the modification of the .md files was correct, but that it was the "clean-blog" theme that was causing the issue. Unfortunately this was the first theme I tried and it doesn't work. Trying another like the "Travelify" theme works fine.

Could you please check this clean-blog theme, as it seems that the index.md contents is not displayed on the site, only the logo is there?
I tried again and did things like described, but still nothing displayed.
You should try installing it fresh from their github source.
That would help.

**Comment by PhrozenByte** [comment]
> Could you please check this clean-blog theme, as it seems that the index.md contents is not displayed on the site, only the logo is there?

Please refer to the theme developer. Custom plugins and themes are made by 3rd parties, unfortunately we can't provide support for this 3rd party content. You should receive help there.

**Comment by gelavat** [comment]
The issue is that almost all themes are not installable nor working out of the box. I think this is not good, because it hurts Pico CMS, whereas it is not your fault.

Themes and plugins functionalities need to have some checks from Pico to make sure that they comply to minimal requirements: must be working, clear and not be implementing hacking systems. You should perform regular little checks (and correction requests to theme providers) about that or at least at each new release of Pico CMS. I posted another issue in this regards.

**Comment by gelavat** [comment]
Hello Maya,

Thank you for your comments.
It is good to know that there was a discussion already about that, I didn't know, but I see that the risk I fear is also shared with others, so it is good :)

Regarding me trying to make noise, it is not the case. I respect a lot the work that you have done you people with Pico CMS. It is just that I like mentioning when I see things not working, rather than just moving to something else, not mentioning what is going wrong like many do. I think it is honorable as it helps the community and can help fix issues and improve the product, and more generally provide an outside feedback that is usually always productive.

Thank you very much for your help proposal to install a theme, it is very nice. And sorry for your real life issues and you still helping me it is very kind. I hope it will resolve soon.
I have been able to figure out what was going wrong for a few of them, trying to ask questions there and providing solutions. So it is fine for what I need, thanks! As you said, themes usually work but we need to tweak things a bit, and this is the part that is difficult for Pico noobs like me.

I provided some thoughts about all that and suggestions in the last comment on issue https://github.com/picocms/Pico/issues/593
I think it could be helping towards this direction quite a lot already.
I suggest continuing on that thread there so it is more related to the subject.

Thanks for your help!

**Comment by mayamcdougall** [comment]
~~I'm only going to say a little here~~, and then continue over on #593. <-- Nevermind, I'm a liar. 😂

First of all, no worries about anything.  I was a little stressed the other day, and I apologize for being blunt. 😅

As long as your intentions are sincere (and you're not just trolling, lol), we absolutely welcome any and all input you want to provide. ❤️

We can't work miracles, there's only a few of us, but we are always open to suggestions.

If you find that you like it here on GitHub, and want to help projects out, here's my suggestion.  If you find old/inaccurate documentation somewhere, in addition to opening an issue, you could always offer to make the changes for them.  This would likely be especially welcomed on some of our Themes, since the original developers have often moved on.

Even Pico itself has a long overdue documentation problem... \*cough\* https://github.com/picocms/picocms.github.io/pull/6, https://github.com/picocms/picocms.github.io/issues/7.

Documentation is hard.  It's difficult and time consuming on its own.  Writing *good* documentation is even harder. 😒

It seems like it should be easy, which in turn makes it easier to be critical of it when something's wrong.  Just remember that often, writing the documentation for your code can be just as complex as the code itself.  And keep in mind that not all contributions to open source need to be *code* contributions.  Helping out with other aspects of projects can be just as valuable. 😉

**Comment by gelavat** [comment]
Hello Maya. Ok no problem it's fine.

Yes I know, doc is not so well liked by many people, but such important. And yes, giving help for only that is already a very good improvement. This is what I am aiming to do.
I provided already an installation and customization section to the clean-blog theme as a PR. It has been accepted and now it is compatible without any issue to install the theme and customize it easily, at least for basic things.

I will try to continue with a few other themes. I think this is the only thing that was really missing, the updated installation instructions and also the customization because of course as soon as we have installed it we want to customize it, change the titles, and so on and then we don't know how to do that.

So this would be the solution I think, to continue updating these installation instructions.
In case the repo owners do not answer, I would suggest we fork the theme and continue to improve it ourselves, and change the address do download the theme. This way we wouldn't be blocked by that.
As these themes are MIT licenced usually, this shouldn't be a problem.

**Comment by mayamcdougall** [comment]
> I think this is the only thing that was really missing

Then I think you need to look at the [Docs](https://picocms.org/docs/) a little closer. 😂

I'm kidding, but they're still a *real* mess in my opinion.

They were supposed to be rewritten *years* ago to be easier for new users to follow... maybe someday we'll finish that. P=

> In case the repo owners do not answer, I would suggest we fork the theme and continue to improve it ourselves

This is *absolutely* a possibility. (It'd just be a lot of work for whoever wants to do it, lol)

The Themes gallery entries each have a freeform `info` property where we specify things like `By:` or `Ported By:` / `Original By:`.  Although it's never happened, if a theme were to be forked, we could easily make it say something like `Originally By:` and `Continued By:` to continue give credit to the original creator while linking to something more up-to-date on the `Learn More` button.

I think in this case if it were *just* an update to documentation, we'd want to see a decent amount of effort put into the fork and not just some typo fixes.  I guess what I mean is that the changes should be *worth* forking for, but it does sound like it could work okay.
